core/server/services/auth/session/store.js

```
@@ -0,0 +1,48 @@
+const {Store} = require('express-session');
+
+module.exports = class SessionStore extends Store {
+    constructor(SessionModel) {
+        super();
+        this.SessionModel = SessionModel;
+    }
+
+    destroy(sid, callback) {
+        this.SessionModel
+            .findOne({session_id: sid})
+            .then((model) => {
+                if (!model) {
+                    return callback(null);
+                }
+                return this.SessionModel
+                    .destroy(model)
+                    .then(() => {
+                        callback(null);
+                    });
+            })
+            .catch(callback);
+    }
+
+    get(sid, callback) {
+        this.SessionModel
+            .findOne({session_id: sid})
+            .then((model) => {
+                if (!model) {
+                    return callback(null, null);
+                }
+                callback(null, model.get('session_data'));
+            })
+            .catch(callback);
+    }
+
+    set(sid, sessionData, callback) {
+        if (!sessionData.user_id) {
+            return callback(new Error('Should not set sessions without a user_id'));
```

Should return a custom error.